Sources/DurationPickerSwiftUI/Public/DurationPickerView.swift

```diff
@@ -0,0 +1,188 @@
+    timeDurationPicker.maximumDuration = maximumDuration
+    
+    timeDurationPicker.addTarget(context.coordinator, action: #selector(Coordinator.changed(_:)), for: .primaryActionTriggered)
+    return timeDurationPicker
+  }
+
+  // This has to be public to comply with UIViewRepresentable, but we don't actually want it to show up in the doc.
+  @_documentation(visibility: internal)
+  public func updateUIView(_ uiView: DurationPicker, context: Context) {
+    if (uiView.duration != duration) {
+      uiView.duration = duration
+    }
+    if (uiView.pickerMode != mode) {
+      uiView.pickerMode = mode
+    }
+    if (uiView.hourInterval != hourInterval) {
+      uiView.hourInterval = hourInterval
+    }
+    if (uiView.minuteInterval != minuteInterval) {
+      uiView.minuteInterval = minuteInterval
+    }
+    if (uiView.secondInterval != secondInterval) {
+      uiView.secondInterval = secondInterval
+    }
+    if (uiView.minimumDuration != minumumDuration) {
+      uiView.minimumDuration = minumumDuration
+    }
+    if (uiView.maximumDuration != maximumDuration) {
+      uiView.maximumDuration = maximumDuration
+    }
+  }
+
+  // This has to be public to comply with UIViewRepresentable, but we don't actually want it to show up in the doc.
+  @_documentation(visibility: internal)
+  public func makeCoordinator() -> DurationPickerView.Coordinator {
+    return Coordinator(duration: $duration)
+  }
+  
+  // This has to be public to comply with UIViewRepresentable, but we don't actually want it to show up in the doc.
+  @_documentation(visibility: internal)
+  public class Coordinator: NSObject {
+    private var duration: Binding<TimeInterval>
+
+    init(duration: Binding<TimeInterval>) {
+      self.duration = duration
+    }
+
+    @objc func changed(_ sender: DurationPicker) {
+      self.duration.wrappedValue = sender.duration
+    }
+  }
+}
+
+@available(iOS 17.0,*)
+#Preview {
+  @Previewable @State var duration: TimeInterval = 60.0 * 30.0
+  @Previewable @State var components: DurationPickerView.Components = .hourMinuteSecond
+  @Previewable @State var minimumDuration: TimeInterval? = nil
+  @Previewable @State var maximumDuration: TimeInterval? = nil
+  @Previewable @State var hourInterval: Int = 1
+  @Previewable @State var minuteInterval: Int = 1
+  @Previewable @State var secondInterval: Int = 1
+
+  // Can't make it case iterable since its defined as an extension.
+  let modes: [DurationPickerView.Components] = [.hour, .hourMinute, .hourMinuteSecond, .minute, .minuteSecond, .second]
+  
+  List {
+    DurationPickerView(
+      $duration,
+      components: components,
+      hourInterval: hourInterval,
+      minuteInterval: minuteInterval,
+      secondInterval: secondInterval,
+      minumumDuration: minimumDuration,
+      maximumDuration: maximumDuration
+    )
+    
+    LabeledContent {
+      Text(Date()..<Date().addingTimeInterval(duration), format: .timeDuration)
```

Comment:
This should be a textfield since we can also set the duration programmatically and have it reflect on the picker